packages/react-components/react-aria/src/AriaLiveAnnouncer/useDomAnnounce.ts

```diff
@@ -144,6 +144,8 @@ export const useDomAnnounce_unstable = (): AriaLiveAnnounceFn => {
     return () => {
       element.remove();
       elementRef.current = null;
+      clearAnnounceTimeout();
+      timeoutRef.current = undefined;
     };
   }, [targetDocument]);
```

Comment:
I think you're missing `clearAnnounceTimeout` in the list of dependencies of `useEffect` here.